modules/core/shared/src/main/scala/scalacache/AbstractCache.scala

```diff
@@ -13,34 +17,36 @@ import scala.language.higherKinds
   *
   * @tparam V The value of types stored in the cache.
   */
-trait AbstractCache[V] extends Cache[V] with LoggingSupport {
+trait AbstractCache[F[_], V] extends Cache[F, V] with LoggingSupport {
 
+  protected implicit def F: MonadError[F, Throwable]
+  protected implicit def Defer: Defer[F]
```

Comment:
For now it looks like we can get away with just a dependency on cats-core.

Reply:
At least until I noticed `Entry` and its timing method (which requires a clock).

Reply:
Yeah. I think a cats-effect dependency in core is fine.

Reply:
Also, I was using Defer for lifting a (most-likely) side-effectful `=> V`, so Sync makes more sense anyway.